src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -190,6 +190,290 @@ function (löhner::IndicatorLöhner)(u::AbstractArray{<:Any,4},
+  @unpack antidiffusive_flux1, antidiffusive_flux2 = cache.ContainerFCT2D
+
+  @unpack alpha, alpha1, alpha2, var_max, var_min, alpha_max_per_element, alpha_mean_per_element = indicator_IDP.cache.ContainerShockCapturingIndicator
+
+  @unpack inverse_weights = dg.basis
+
+  @unpack alpha_maxIDP = indicator_IDP
+
+  @threaded for element in eachelement(dg, cache)
+
+    # Calculate indicator variables at Gauss-Lobatto nodes
+    indicator = indicator_threaded[Threads.threadid()]
+    for j in eachnode(dg), i in eachnode(dg)
+      u_local = get_node_vars(u_old, equations, dg, i, j, element)
+      indicator[i, j] = indicator_IDP.variable(u_local, equations)
+    end
+
+    for j in eachnode(dg), i in eachnode(dg)
+      # Calculate max and min of variable at Gauss-Lobatto nodes
+      var_min[i, j, element] = indicator[i, j]
+      var_max[i, j, element] = indicator[i, j]
+      if i > 1
+        var_min[i, j, element] = min(var_min[i, j, element], indicator[i-1, j])
+        var_max[i, j, element] = max(var_max[i, j, element], indicator[i-1, j])
+      end
+      if i < nnodes(dg)
+        var_min[i, j, element] = min(var_min[i, j, element], indicator[i+1, j])
+        var_max[i, j, element] = max(var_max[i, j, element], indicator[i+1, j])
+      end
+      if j > 1
+        var_min[i, j, element] = min(var_min[i, j, element], indicator[i, j-1])
+        var_max[i, j, element] = max(var_max[i, j, element], indicator[i, j-1])
+      end
+      if j < nnodes(dg)
+        var_min[i, j, element] = min(var_min[i, j, element], indicator[i, j+1])
+        var_max[i, j, element] = max(var_max[i, j, element], indicator[i, j+1])
+      end
+    end
+  end
+
+  # Loop over interfaces
+  for interface in eachinterface(dg, cache)
+    # Get neighboring element ids
+    left  = cache.interfaces.neighbor_ids[1, interface]
+    right = cache.interfaces.neighbor_ids[2, interface]
+
+    orientation = cache.interfaces.orientations[interface]
+
+    for i in eachnode(dg)
+      if orientation == 1
+        index_left  = (nnodes(dg), i, left)
+        index_right = (1,          i, right)
+      else
+        index_left  = (i, nnodes(dg), left)
+        index_right = (i,          1, right)
+      end
+      u_local_left  = get_node_vars(u_old, equations, dg, index_left...)
+      u_local_right = get_node_vars(u_old, equations, dg, index_right...)
+      var_neighbor_left  = indicator_IDP.variable(u_local_left,  equations)
+      var_neighbor_right = indicator_IDP.variable(u_local_right, equations)
+
+      var_min[index_right...] = min(var_min[index_right...], var_neighbor_left)
+      var_max[index_right...] = max(var_max[index_right...], var_neighbor_left)
+
+      var_min[index_left...] = min(var_min[index_left...], var_neighbor_right)
+      var_max[index_left...] = max(var_max[index_left...], var_neighbor_right)
+    end
+  end
+
+  # Loop over L2 mortars
+  for mortar in eachmortar(dg, cache)
+    # Get neighboring element ids
```

Comment:
@ranocha I think the code is quite understandable. But as I mentioned, the mortar iteration got a bit long since I have almost the exact same code for four cases (large element on the left, right, bottom, top). Do you see any possibility to shorten this stuff? [Here](https://github.com/bennibolm/Trixi.jl/blob/staggered-grid-volume-integral/src/solvers/dgsem_tree/indicators_2d.jl#L285-L419) is the complete code.